Sources/SmithyChecksums/ChunkedStream.swift

```diff
@@ -55,68 +56,56 @@ class AWSChunkedStream {
         self.signingConfig = signingConfig
         self.previousSignature = previousSignature
         self.trailingHeaders = trailingHeaders
+        self.checksumString = checksumString
 
-        // Determine the checksum algorithm, if provided
-        let checksumAlgorithm: ChecksumAlgorithm?
-        if let checksumString = checksumString {
-            checksumAlgorithm = ChecksumAlgorithm.from(string: checksumString)
-
-            // Unsupported checksum
-            if checksumAlgorithm == nil {
-                throw UnknownChecksumError.notSupported(checksum: checksumString)
-            }
-        } else {
-            checksumAlgorithm = nil
-        }
-
-        self.chunkedReader = AWSChunkedReader(
```

Comment:
this code has been moved from stream creation to the reader creation which happens on init for dependency management + we do not need the `ChecksumAlgorithm` yet when we're creating the Stream type